### ISSUE TITLE
App Crash: Faulting module name: ucrtbase.dll

### ISSUE THREAD

**Issue by NickSebClark** [pal1000/mesa-dist-win#162]
Hi

I've installed 23.1.9 using the system wide deploy tool on Windows Server 2022.

When I try to run my app it immediately crashes.

Event viewer shows:

Faulting module name: ucrtbase.dll, version: 10.0.20348.1, time stamp: 0xdacb234c
Exception code: 0xc0000409

Any ideas?

Other attempts:

MinGW version 23.2.1 per-app-deployment
This results in zlib1.dll not found.
I've tried llvmpipe and softpipe

**Comment by NickSebClark** [comment]
I realised that prehaps this is the wrong version.

I changed to MinGW version on 23.2.1.

Now my app reports that OpenGL is not installed

**Comment by pal1000** [comment]
> Hi
> 
> I've installed 23.1.9 using the system wide deploy tool on Windows Server 2022.
> 
> When I try to run my app it immediately crashes.
> 
> Event viewer shows:
> 
> Faulting module name: ucrtbase.dll, version: 10.0.20348.1, time stamp: 0xdacb234c Exception code: 0xc0000409
> 
> Any ideas?

Please try 23.2.1 and 23.3.0-rc3 MSVC builds. 23.1 branch is EOL so it will never be fixed. If you still have problems you need to collect a call stack. Debug symbols are in MSVC debug info package.

> Other attempts:
> 
> MinGW version 23.2.1 per-app-deployment This results in zlib1.dll not found. I've tried llvmpipe and softpipe

This is odd. I need to run some tests to see if I can reproduce this error.

**Comment by pal1000** [comment]
> > Other attempts:
> > MinGW version 23.2.1 per-app-deployment This results in zlib1.dll not found. I've tried llvmpipe and softpipe
> 
> This is odd. I need to run some tests to see if I can reproduce this error.

I was able to reproduce but I couldn't figure out why this is happening. I made no changes to link flags since at least [23.1.2](https://github.com/pal1000/mesa-dist-win/compare/23.1.2...master#diff-6e032038c5857226d6c2b00eb9378c89487061442ff8e90b6197feac93346f17) when MinGW binaries were working decently.

**Comment by NickSebClark** [comment]
So to confirm we think the MinGW build is just broken and we have to wait for it to be fixed?

**Comment by pal1000** [comment]
Yes and I think it will stay broken until [this passes review and gets merged](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/24486).
